cl/corpus_importer/bulk_utils.py

```diff
@@ -27,9 +28,11 @@ def docket_pks_for_query(query_string):
     )
     main_query["group.limit"] = 0
     main_query["sort"] = "dateFiled asc"
-    si = ExtraSolrInterface(settings.SOLR_RECAP_URL, mode="r")
-    search = si.query().add_extra(**main_query)
-    si.conn.http_connection.close()
+    with Session() as session:
+        si = ExtraSolrInterface(
+            settings.SOLR_RECAP_URL, http_connection=session, mode="r"
+        )
+        search = si.query().add_extra(**main_query)
     page_size = 100
```

Comment:
I worry a bit here. If the session is closed before we paginate, are we sure it would work? I suspect that the paginator triggers the actual query, so it might fail if the session is closed? 

I'm not sure if we have a test for this function, but if not, it might be nice to have. It's only used in a few bulk import scripts though, so if we just test by hand, that's OK too.

Reply:
> I worry a bit here. If the session is closed before we paginate, are we sure it would work? I suspect that the paginator triggers the actual query, so it might fail if the session is closed?

I think I was just making sure usages of `si` were inside the session block. I think that should be sufficient to ensure it works.

Reply:
> I'm not sure if we have a test for this function, but if not, it might be nice to have. It's only used in a few bulk import scripts though, so if we just test by hand, that's OK too.

Since this previously used an explicit close() it shouldn't cause regressions as I'm effectively doing the same thing but with a context manager.

Reply:
OK, good point. I'm surprised it works as is. Weird.